pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11981022?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11981022?buildTab=artifacts#/) on master @ [27e7c166b5591b0e05bb2fd23780309b490d7af9](https://github.com/cockroachdb/cockroach/commits/27e7c166b5591b0e05bb2fd23780309b490d7af9):


```
pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.
[07:01:16] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3188/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 11 statements
    logic.go:2576: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3188/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:01:17] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3188/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3993: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3188/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3993: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3188/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed239503891
--- FAIL: TestCCLLogic_regional_by_row_read_committed (29.31s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31919

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/11994285?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/11994285?buildTab=artifacts#/) on master @ [fad649d89721ddb3e9f3dcab1ad5d14f74c91bc9](https://github.com/cockroachdb/cockroach/commits/fad649d89721ddb3e9f3dcab1ad5d14f74c91bc9):


```
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.
[05:30:21] setting distsql_workmem='92534B';
[05:30:22] setting distsql_workmem='92534B';
    logic.go:2576: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3980/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[05:30:24] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3980/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3993: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3980/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3993: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3980/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/24b232bb2ddb5cbd23310506078b09b9/logTestCCLLogic_regional_by_row_read_committed1093372550
--- FAIL: TestCCLLogic_regional_by_row_read_committed (21.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

This test also fails under stress at the commit prior to d0c46fe66ce86a4820fca54b4983bdb95f838c84: 4b2badd7bc6c11a7646e66b131f5bcaf8328916a, but with a different error:
```
dev test pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant --filter=TestCCLLogic_regional_by_row_read_committed --stress
```
```
FAIL: //pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant:multiregion-9node-3region-3azs-tenant_test (run 828 of 1000) (see /private/var/tmp/_bazel_msirek/70437e77fb93b11771e5e34486b10eac/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/testlogs/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test/run_828_of_1000/test.log)
ERROR: /Users/msirek/go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/BUILD.bazel:3:8: Testing //pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant:multiregion-9node-3region-3azs-tenant_test (run 828 of 1000) failed: Test failed, aborting
==================== Test output for //pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant:multiregion-9node-3region-3azs-tenant_test (run 828 of 1000):
I230929 23:57:42.423516 1 (gostd) rand.go:250  [-] 1  random seed: -2889133837769026593

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1558: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.
test logs left over in: /Users/msirek/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/6de158fee3dfcf3c1fb1427c6ae12320/logTestCCLLogic_regional_by_row_read_committed2329509628
E230930 00:04:58.895208 40531 2@rpc/peer.go:580  [T10,Vtenant-10,nsql5,rnode=1,raddr=127.0.0.1:58327,class=default,rpc] 2  disconnected (was healthy for 4.055s): grpc: connection error: desc = "transport: error while dialing: connection interrupted (did the remote node shut down or are there networking issues?)" [code 14/Unavailable]
E230930 00:04:58.990365 44281 2@rpc/peer.go:580  [T10,Vtenant-10,nsql8,rnode=1,raddr=127.0.0.1:58327,class=default,rpc] 3  disconnected (was healthy for 3.025s): grpc: connection error: desc = "transport: error while dialing: connection interrupted (did the remote node shut down or are there networking issues?)" [code 14/Unavailable]
I230930 00:04:59.162660 50820 sql/sqlliveness/slstorage/slstorage.go:394  [n3,intExec=expire-sessions] 4  deleted session 0101801b00ff1b6c7f422d9013d7738bf4f5d6 which expired at 1696031907.752038000,0
--- FAIL: TestCCLLogic_regional_by_row_read_committed (9.86s)
    test_log_scope.go:167: test logs captured to: /Users/msirek/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/6de158fee3dfcf3c1fb1427c6ae12320/logTestCCLLogic_regional_by_row_read_committed2329509628
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:1656:
        :0: error while processing
    logic.go:1656: dial tcp 127.0.0.1:58325: connect: connection refused
    panic.go:522: -- test log scope end --
FAIL
I230930 00:04:59.222348 1 (gostd) testmain.go:118  [-] 1  Test //pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant:multiregion-9node-3region-3azs-tenant_test exited with error code 1
================================================================================
Target //pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant:multiregion-9node-3region-3azs-tenant_test up-to-date:
  _bazel/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test
ERROR: Test failed, aborting
INFO: Elapsed time: 2939.185s, Critical Path: 95.03s
INFO: 502 processes: 114 remote cache hit, 13 internal, 375 darwin-sandbox.
//pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant:multiregion-9node-3region-3azs-tenant_test NO STATUS
  /private/var/tmp/_bazel_msirek/70437e77fb93b11771e5e34486b10eac/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin_arm64-fastbuild/testlogs/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test/run_828_of_1000/test.log
```

d0c46fe66ce86a4820fca54b4983bdb95f838c84 added the logic test that is failing, but I don't think the failure is related to read committed isolation. I'm able to reproduce it using this smaller logic test which runs entirely at serializable isolation:

```
# michae2@michae2-crl-mbp 1 (1) ~crdb % cat pkg/ccl/logictestccl/testdata/logic_test/unique

# tenant-cluster-setting-override-opt: sql.zone_configs.allow_for_secondary_tenant.enabled=true sql.multi_region.allow_abstractions_for_secondary_tenants.enabled=true
# LogicTest: multiregion-9node-3region-3azs multiregion-9node-3region-3azs-vec-off multiregion-9node-3region-3azs-tenant multiregion-9node-3region-3azs-no-los

statement ok
CREATE DATABASE multi_region_test_db PRIMARY REGION "ca-central-1" REGIONS "ap-southeast-2", "us-east-1" SURVIVE REGION FAILURE

statement ok
USE multi_region_test_db

statement ok
CREATE TABLE ab (
  region crdb_internal_region NOT NULL,
  a STRING NOT NULL,
  b STRING NOT NULL,
  PRIMARY KEY (region, a),
  UNIQUE INDEX (region, b),
  FAMILY (region, a, b)
) LOCALITY REGIONAL BY ROW AS region

statement ok
INSERT INTO ab VALUES ('ap-southeast-2', 'aaa', 'bbb'), ('ca-central-1', 'ccc', 'ddd')

statement error pgcode 23505 pq: duplicate key value violates unique constraint "ab_region_b_key"\nDETAIL: Key \(region, b\)=\('ca-central-1', 'ddd'\) already exists
UPDATE ab SET region = 'ca-central-1', b = 'ddd' WHERE a = 'aaa'

# michae2@michae2-crl-mbp 146 (1) ~crdb % ./dev testlogic ccl --config=multiregion-9node-3region-3azs-tenant --files=unique --ignore-cache --stress
...
--- FAIL: TestCCLLogic_unique (16.53s)
    test_log_scope.go:167: test logs captured to: /Users/michae2/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/59032834933fdc70df3058d7ece5f2d7/logTestCCLLogic_unique2408354390
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:2576:

        /private/var/tmp/_bazel_michae2/239414c2d9c0d0876d35ed2ff4c5fe74/sandbox/darwin-sandbox/372/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/unique:23:
        expected:
        pq: duplicate key value violates unique constraint "ab_region_b_key"\nDETAIL: Key \(region, b\)=\('ca-central-1', 'ddd'\) already exists
        got:
        pq: duplicate key value got decoding error, key=/Min val=raw_bytes:"\357\340\022\215\003\022ccc\000\001" timestamp:<wall_time:1696196042593730000 > : insufficient bytes to decode uvarint value
    logic.go:4172: -- test log scope end --
FAIL
```

Notice the slightly changed error message, it always seems to happen when key is `/Min`. And it doesn't seem to reproduce when using `--config=multiregion-9node-3region-3azs`, only with `--config=multiregion-9node-3region-3azs-tenant` so I suspect that both regional by row tables and multi-tenancy are required. I will keep investigating.

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12009532?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12009532?buildTab=artifacts#/) on master @ [f6f355b50e0dbf28633e25ddd05f2775141af31e](https://github.com/cockroachdb/cockroach/commits/f6f355b50e0dbf28633e25ddd05f2775141af31e):


```
pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.

pkg/sql/logictest/logic.go:1556: ((*logicTest).newCluster)
	NOTICE: .StartTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartTenant() instead.
[06:59:21] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3189/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 7 statements
    logic.go:2576: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3189/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[06:59:23] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3189/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3993: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3189/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3993: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3189/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed131772037
--- FAIL: TestCCLLogic_regional_by_row_read_committed (34.44s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12026932?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12026932?buildTab=artifacts#/) on master @ [4494e945a9070862cfae1b668b3f8b667c41be17](https://github.com/cockroachdb/cockroach/commits/4494e945a9070862cfae1b668b3f8b667c41be17):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed302651180
    test_log_scope.go:81: use -show-logs to present logs inline
[07:25:34] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3183/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 6 statements
    logic.go:2578: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3183/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:25:35] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3183/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3995: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3183/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3995: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3183/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed302651180
--- FAIL: TestCCLLogic_regional_by_row_read_committed (37.49s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12043006?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12043006?buildTab=artifacts#/) on master @ [1b83c04ed0106dd0d3380821707861008ec32b73](https://github.com/cockroachdb/cockroach/commits/1b83c04ed0106dd0d3380821707861008ec32b73):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed310769343
    test_log_scope.go:81: use -show-logs to present logs inline
[07:22:40] setting distsql_workmem='62209B';
[07:22:40] setting distsql_workmem='62209B';
[07:22:43] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2819/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 8 statements
    logic.go:2578: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2819/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:22:44] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2819/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3995: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2819/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3995: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2819/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed310769343
--- FAIL: TestCCLLogic_regional_by_row_read_committed (27.30s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Coverage_UnitTestsCCL/12080469?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Coverage_UnitTestsCCL/12080469?buildTab=artifacts#/) on master @ [3b438b4a59ad4759e5bf22ff0b6dd6e678c2be0d](https://github.com/cockroachdb/cockroach/commits/3b438b4a59ad4759e5bf22ff0b6dd6e678c2be0d):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/24b232bb2ddb5cbd23310506078b09b9/logTestCCLLogic_regional_by_row_read_committed827710860
    test_log_scope.go:81: use -show-logs to present logs inline
[15:39:37] setting distsql_workmem='82227B';
[15:39:37] setting distsql_workmem='82227B';
    logic.go:2578: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4916/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[15:39:39] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4916/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 32 statements
[15:39:39] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4916/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3995: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4916/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3995: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4916/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/24b232bb2ddb5cbd23310506078b09b9/logTestCCLLogic_regional_by_row_read_committed827710860
--- FAIL: TestCCLLogic_regional_by_row_read_committed (20.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12122356?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12122356?buildTab=artifacts#/) on master @ [7e1f909e9f2ddcef2e418d6c2df11f3139a2b85a](https://github.com/cockroachdb/cockroach/commits/7e1f909e9f2ddcef2e418d6c2df11f3139a2b85a):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2490017701
    test_log_scope.go:81: use -show-logs to present logs inline
[06:54:24] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3031/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 8 statements
    logic.go:2578: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3031/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[06:54:26] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3031/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3995: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3031/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3995: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3031/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2490017701
--- FAIL: TestCCLLogic_regional_by_row_read_committed (27.83s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12137876?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12137876?buildTab=artifacts#/) on master @ [5af56c0ecd701d452c19fb89908012294b5336aa](https://github.com/cockroachdb/cockroach/commits/5af56c0ecd701d452c19fb89908012294b5336aa):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2266644065
    test_log_scope.go:81: use -show-logs to present logs inline
[07:15:30] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3314/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 12 statements
    logic.go:2578: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3314/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:15:31] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3314/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3995: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3314/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3995: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3314/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2266644065
--- FAIL: TestCCLLogic_regional_by_row_read_committed (26.60s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12149590?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12149590?buildTab=artifacts#/) on master @ [619aa6edad0269f75d12cb7d4b88b52f09825c7b](https://github.com/cockroachdb/cockroach/commits/619aa6edad0269f75d12cb7d4b88b52f09825c7b):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2820547160
    test_log_scope.go:81: use -show-logs to present logs inline
[07:57:44] setting distsql_workmem='78741B';
[07:57:44] setting distsql_workmem='78741B';
[07:57:47] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3216/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 7 statements
[07:57:49] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3216/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 22 statements
    logic.go:2578: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3216/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:57:50] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3216/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3995: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3216/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3995: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3216/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2820547160
--- FAIL: TestCCLLogic_regional_by_row_read_committed (35.69s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112160 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/12211529?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_SqlLogicDeclarativeSchemaChangerCorpusBazel/12211529?buildTab=artifacts#/) on master @ [45f6344a2fc7d2d52f410e1ecad98f8c928cfeea](https://github.com/cockroachdb/cockroach/commits/45f6344a2fc7d2d52f410e1ecad98f8c928cfeea):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/24b232bb2ddb5cbd23310506078b09b9/logTestCCLLogic_regional_by_row_read_committed1879932226
    test_log_scope.go:81: use -show-logs to present logs inline
[05:32:45] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3978/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 12 statements
    logic.go:2579: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3978/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value violates unique constraint "river_pkey"
        DETAIL: Key (region, name)=('us-east-1', 'Skykomish') already exists.
[05:32:45] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3978/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3996: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3978/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3996: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3978/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/24b232bb2ddb5cbd23310506078b09b9/logTestCCLLogic_regional_by_row_read_committed1879932226
--- FAIL: TestCCLLogic_regional_by_row_read_committed (24.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112160 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12212546?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12212546?buildTab=artifacts#/) on master @ [45f6344a2fc7d2d52f410e1ecad98f8c928cfeea](https://github.com/cockroachdb/cockroach/commits/45f6344a2fc7d2d52f410e1ecad98f8c928cfeea):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2813368806
    test_log_scope.go:81: use -show-logs to present logs inline
[07:32:58] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3351/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 11 statements
    logic.go:2579: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3351/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:32:59] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3351/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3996: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3351/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3996: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3351/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed2813368806
--- FAIL: TestCCLLogic_regional_by_row_read_committed (29.05s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112160 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCCLLogic_regional_by_row_read_committed2596540546
    test_log_scope.go:81: use -show-logs to present logs inline
[17:32:51] setting distsql_workmem='98203B';
[17:32:51] setting distsql_workmem='98203B';
    logic.go:2581: 
         
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[17:32:52] --- done: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_regional_by_row_read_committed2596540546
--- FAIL: TestCCLLogic_regional_by_row_read_committed (11.01s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=10</code>
, <code>shard=15</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112746 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCCLLogic_regional_by_row_read_committed3428247824
    test_log_scope.go:81: use -show-logs to present logs inline
[17:32:54] setting distsql_workmem='30036B';
[17:32:54] setting distsql_workmem='30036B';
    logic.go:2581: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[17:32:56] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_regional_by_row_read_committed3428247824
--- FAIL: TestCCLLogic_regional_by_row_read_committed (13.42s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=16</code>
, <code>shard=15</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112746 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCCLLogic_regional_by_row_read_committed127580084
    test_log_scope.go:81: use -show-logs to present logs inline
[17:32:51] setting distsql_workmem='75373B';
[17:32:51] setting distsql_workmem='75373B';
[17:32:54] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 26 statements
    logic.go:2581: 
         
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[17:32:54] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_regional_by_row_read_committed127580084
--- FAIL: TestCCLLogic_regional_by_row_read_committed (13.17s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=9</code>
, <code>shard=15</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112746 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://tanzanite.cluster.engflow.com/invocation/817e8f2d-e4dc-4b32-bfbb-783ac3877f53) on master @ [59a69ae0c5474a8b323afac269055bf6ce8afa34](https://github.com/cockroachdb/cockroach/commits/59a69ae0c5474a8b323afac269055bf6ce8afa34):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCCLLogic_regional_by_row_read_committed970791551
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:2581: 
         
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[17:31:15] --- done: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_regional_by_row_read_committed970791551
--- FAIL: TestCCLLogic_regional_by_row_read_committed (11.02s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=7</code>
, <code>shard=15</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112746 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://tanzanite.cluster.engflow.com/invocation/0b6ce399-bae7-4955-9f35-d99a012daeb3) on master @ [bf9a22dc85683966ee65e84d3eeadf2b44185127](https://github.com/cockroachdb/cockroach/commits/bf9a22dc85683966ee65e84d3eeadf2b44185127):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCCLLogic_regional_by_row_read_committed186029346
    test_log_scope.go:81: use -show-logs to present logs inline
[21:19:36] setting distsql_workmem='95462B';
[21:19:36] setting distsql_workmem='95462B';
    logic.go:2581: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[21:19:37] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_regional_by_row_read_committed186029346
--- FAIL: TestCCLLogic_regional_by_row_read_committed (10.16s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=11</code>
, <code>shard=15</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112833 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12341291?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12341291?buildTab=artifacts#/) on master @ [8315a4bc997fb8b8d679079e14d6d7ca94d53bc6](https://github.com/cockroachdb/cockroach/commits/8315a4bc997fb8b8d679079e14d6d7ca94d53bc6):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed759714094
    test_log_scope.go:81: use -show-logs to present logs inline
[07:49:05] setting distsql_workmem='50442B';
[07:49:06] setting distsql_workmem='50442B';
[07:49:08] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3279/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 7 statements
    logic.go:2581: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3279/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:49:09] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3279/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3279/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3279/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
--- FAIL: TestCCLLogic_regional_by_row_read_committed (44.51s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112833 pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test: TestCCLLogic_regional_by_row_read_committed failed [C-test-failure O-robot T-sql-queries branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/12360254?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/12360254?buildTab=artifacts#/) on master @ [2205334339583cea57db33e3c0d992f672e660e8](https://github.com/cockroachdb/cockroach/commits/2205334339583cea57db33e3c0d992f672e660e8):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/24b232bb2ddb5cbd23310506078b09b9/logTestCCLLogic_regional_by_row_read_committed2888135195
    test_log_scope.go:81: use -show-logs to present logs inline
[06:17:01] setting distsql_workmem='42930B';
[06:17:01] setting distsql_workmem='42930B';
[06:17:03] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5286/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 11 statements
    logic.go:2581: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5286/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[06:17:03] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5286/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5286/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5286/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/24b232bb2ddb5cbd23310506078b09b9/logTestCCLLogic_regional_by_row_read_committed2888135195
--- FAIL: TestCCLLogic_regional_by_row_read_committed (19.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test.TestCCLLogic_regional_by_row_read_committed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12447519?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12447519?buildTab=artifacts#/) on master @ [bfc5c35b112fe612d837ef247b2c61e532e8184d](https://github.com/cockroachdb/cockroach/commits/bfc5c35b112fe612d837ef247b2c61e532e8184d):


```
=== RUN   TestCCLLogic_regional_by_row_read_committed
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0451cbd40094c53fa53acd1ee2497281/logTestCCLLogic_regional_by_row_read_committed3389879511
    test_log_scope.go:81: use -show-logs to present logs inline
[07:47:56] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3282/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 7 statements
[07:47:59] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3282/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed: 23 statements
    logic.go:2581: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3282/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:180: 
        expected:
        pq: duplicate key value violates unique constraint "river_region_outlet_key"\nDETAIL: Key \(region, outlet\)=\('us-east-1', 'Salish Sea'\) already exists\.
        got:
        pq: duplicate key value got decoding error: insufficient bytes to decode uvarint value
[07:48:00] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3282/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed with config multiregion-9node-3region-3azs-tenant: 32 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3282/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3282/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-tenant/multiregion-9node-3region-3azs-tenant_test_/multiregion-9node-3region-3azs-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row_read_committed:183: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
--- FAIL: TestCCLLogic_regional_by_row_read_committed (45.40s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_regional_by_row_read_committed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/kvcoord/dist_sender.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender.go
@@ -1129,7 +1129,11 @@ func (ds *DistSender) incrementBatchCounters(ba *kvpb.BatchRequest) {
 }
 
 type response struct {
-	reply     *kvpb.BatchResponse
+	reply *kvpb.BatchResponse
+	// positions argument describes how the given batch request corresponds to
+	// the original, un-truncated one, and allows us to combine the response
+	// later via BatchResponse.Combine. (nil positions should be used when the
+	// original batch request is fully contained within a single range.)
 	positions []int
 	pErr      *kvpb.Error
 }
@@ -1449,8 +1453,10 @@ func (ds *DistSender) divideAndSendBatchToRanges(
 	// Take the fast path if this batch fits within a single range.
 	if !ri.NeedAnother(rs) {
 		resp := ds.sendPartialBatch(
-			ctx, ba, rs, isReverse, withCommit, batchIdx, ri.Token(), nil, /* positions */
+			ctx, ba, rs, isReverse, withCommit, batchIdx, ri.Token(),
 		)
+		// resp.positions remains nil since the original batch is fully
+		// contained within a single range.
 		return resp.reply, resp.pErr
 	}
 
@@ -1534,6 +1540,11 @@ func (ds *DistSender) divideAndSendBatchToRanges(
 		for _, responseCh := range responseChs {
 			resp := <-responseCh
 			if resp.pErr != nil {
+				// Re-map the error index within this partial batch back to its
+				// position in the encompassing batch.
+				if resp.pErr.Index != nil && resp.pErr.Index.Index != -1 && resp.positions != nil {
+					resp.pErr.Index.Index = int32(resp.positions[resp.pErr.Index.Index])
+				}
 				if pErr == nil {
 					pErr = resp.pErr
 					// Update the error's transaction with any new information from
@@ -1647,8 +1658,9 @@ func (ds *DistSender) divideAndSendBatchToRanges(
 			// Sent the batch asynchronously.
 		} else {
 			resp := ds.sendPartialBatch(
-				ctx, curRangeBatch, curRangeRS, isReverse, withCommit, batchIdx, ri.Token(), positions,
+				ctx, curRangeBatch, curRangeRS, isReverse, withCommit, batchIdx, ri.Token(),
 			)
+			resp.positions = positions
 			responseCh <- resp
 			if resp.pErr != nil {
 				return
@@ -1749,9 +1761,9 @@ func (ds *DistSender) sendPartialBatchAsync(
 		},
 		func(ctx context.Context) {
 			ds.metrics.AsyncSentCount.Inc(1)
-			responseCh <- ds.sendPartialBatch(
-				ctx, ba, rs, isReverse, withCommit, batchIdx, routing, positions,
-			)
+			resp := ds.sendPartialBatch(ctx, ba, rs, isReverse, withCommit, batchIdx, routing)
+			resp.positions = positions
+			responseCh <- resp
 		},
 	); err != nil {
 		ds.metrics.AsyncThrottledCount.Inc(1)
@@ -1789,11 +1801,7 @@ func slowRangeRPCReturnWarningStr(s *redact.StringBuilder, dur time.Duration, at
 // request are limited to the range's key span. The rs argument corresponds to
 // the span encompassing the key ranges of all requests in the truncated batch.
 // It should be entirely contained within the range descriptor of the supplied
-// routing token. The positions argument describes how the given batch request
-// corresponds to the original, un-truncated one, and allows us to combine the
-// response later via BatchResponse.Combine. (nil positions argument should be
-// used when the original batch request is fully contained within a single
-// range.)
+// routing token.
 //
 // The send occurs in a retry loop to handle send failures. On failure to send
 // to any replicas, we backoff and retry by refetching the range descriptor. If
@@ -1809,7 +1817,6 @@ func (ds *DistSender) sendPartialBatch(
 	withCommit bool,
 	batchIdx int,
 	routingTok rangecache.EvictionToken,
-	positions []int,
 ) response {
 	if batchIdx == 1 {
 		ds.metrics.PartialBatchCount.Inc(2) // account for first batch
@@ -1872,7 +1879,7 @@ func (ds *DistSender) sendPartialBatch(
 			if !intersection.Equal(rs) {
 				log.Eventf(ctx, "range shrunk; sub-dividing the request")
 				reply, pErr = ds.divideAndSendBatchToRanges(ctx, ba, rs, isReverse, withCommit, batchIdx)
-				return response{reply: reply, positions: positions, pErr: pErr}
+				return response{reply: reply, pErr: pErr}
 			}
 		}
 
@@ -1930,18 +1937,12 @@ func (ds *DistSender) sendPartialBatch(
 
 		// If sending succeeded, return immediately.
 		if reply.Error == nil {
-			return response{reply: reply, positions: positions}
+			return response{reply: reply}
 		}
 
 		// Untangle the error from the received response.
 		pErr = reply.Error
 		reply.Error = nil // scrub the response error
-
-		// Re-map the error index within this partial batch back
-		// to its position in the encompassing batch.
-		if pErr.Index != nil && pErr.Index.Index != -1 && positions != nil {
-			pErr.Index.Index = int32(positions[pErr.Index.Index])
-		}
 
 		log.VErrEventf(ctx, 2, "reply error %s: %s", ba, pErr)
 
@@ -1979,7 +1980,7 @@ func (ds *DistSender) sendPartialBatch(
 			// with unknown mapping to our truncated reply).
 			log.VEventf(ctx, 1, "likely split; will resend. Got new descriptors: %s", tErr.Ranges)
 			reply, pErr = ds.divideAndSendBatchToRanges(ctx, ba, rs, isReverse, withCommit, batchIdx)
-			return response{reply: reply, positions: positions, pErr: pErr}
+			return response{reply: reply, pErr: pErr}
 		}
 		break
 	}

--- a/pkg/kv/kvclient/kvcoord/dist_sender_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_test.go
@@ -1663,9 +1663,9 @@ func TestEvictCacheOnError(t *testing.T) {
 		},
 	}
 
-	rangeMismachErr := kvpb.NewRangeKeyMismatchError(
+	rangeMismatchErr := kvpb.NewRangeKeyMismatchError(
 		context.Background(), nil, nil, &lhs, nil /* lease */)
-	rangeMismachErr.AppendRangeInfo(context.Background(), roachpb.RangeInfo{Desc: rhs, Lease: roachpb.Lease{}})
+	rangeMismatchErr.AppendRangeInfo(context.Background(), roachpb.RangeInfo{Desc: rhs, Lease: roachpb.Lease{}})
 
 	testCases := []struct {
 		canceledCtx            bool
@@ -1674,7 +1674,7 @@ func TestEvictCacheOnError(t *testing.T) {
 		shouldClearReplica     bool
 	}{
 		{false, errors.New(errString), false, false},      // non-retryable replica error
-		{false, rangeMismachErr, false, false},            // RangeKeyMismatch replica error
+		{false, rangeMismatchErr, false, false},           // RangeKeyMismatch replica error
 		{false, &kvpb.RangeNotFoundError{}, false, false}, // RangeNotFound replica error
 		{false, nil, false, false},                        // RPC error
 		{true, nil, false, false},                         // canceled context
@@ -4550,6 +4550,169 @@ func TestDistSenderSlowLogMessage(t *testing.T) {
 		act := s.RedactableString()
 		require.EqualValues(t, exp, act)
 	}
+}
+
+// TestErrorIndexOnRangeSplit verifies that in case divideAndSendBatchToRanges
+// is called recursively due to a stale range descriptor and the error is
+// encountered during that recursive call, the error index is set correctly to
+// point to the request in the original batch. It is a regression test for
+// #111481.
+func TestErrorIndexOnRangeSplit(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	keyA := roachpb.Key("a")
+	keyB := roachpb.Key("b")
+	keyC := roachpb.Key("c")
+	splitKey := keys.MustAddr(keyC)
+	newSplitKey := keys.MustAddr(keyB)
+
+	ctx := context.Background()
+	tr := tracing.NewTracer()
+	stopper := stop.NewStopper(stop.WithTracer(tr))
+	defer stopper.Stop(ctx)
+
+	clock := hlc.NewClockForTesting(nil)
+	rpcContext := rpc.NewInsecureTestingContext(ctx, clock, stopper)
+	g := makeGossip(t, stopper, rpcContext)
+
+	// We start out with two ranges split at splitKey. This is needed so that
+	// we don't hit the fast-path in divideAndSendBatchToRanges where only a
+	// single range is touched.
+	repls := []roachpb.ReplicaDescriptor{
+		{
+			NodeID:  1,
+			StoreID: 1,
+		},
+		{
+			NodeID:  1,
+			StoreID: 1,
+		},
+	}
+	initDescs := []roachpb.RangeDescriptor{
+		{
+			RangeID:          roachpb.RangeID(1),
+			Generation:       1,
+			StartKey:         roachpb.RKeyMin,
+			EndKey:           splitKey,
+			InternalReplicas: repls,
+		},
+		{
+			RangeID:          roachpb.RangeID(2),
+			Generation:       1,
+			StartKey:         splitKey,
+			EndKey:           roachpb.RKeyMax,
+			InternalReplicas: repls,
+		},
+	}
+	// We'll send a BatchRequest touching the original two ranges, which is so
+	// that it is split into two parallel BatchRequests, then on the one
+	// touching the first range we'll inject a range key mismatch error to
+	// simulate a range split. Thus, after the injected split we'll have three
+	// ranges with split points at newSplitKey and splitKey.
+	splitDescs := []roachpb.RangeDescriptor{
+		{
+			RangeID:          roachpb.RangeID(1),
+			Generation:       2,
+			StartKey:         roachpb.RKeyMin,
+			EndKey:           newSplitKey,
+			InternalReplicas: repls,
+		},
+		{
+			RangeID:          roachpb.RangeID(3),
+			Generation:       2,
+			StartKey:         newSplitKey,
+			EndKey:           splitKey,
+			InternalReplicas: repls,
+		},
+		{
+			RangeID:          roachpb.RangeID(2),
+			Generation:       2,
+			StartKey:         splitKey,
+			EndKey:           roachpb.RKeyMax,
+			InternalReplicas: repls,
+		},
+	}
+
+	initialRDB := mockRangeDescriptorDBForDescs(initDescs...)
+	splitRDB := mockRangeDescriptorDBForDescs(splitDescs...)
+
+	var rc *rangecache.RangeCache
+	var wg sync.WaitGroup
+	var once sync.Once
+	wg.Add(2)
+	waitThenSwitchToSplitDesc := func() {
+		// Wait for both partial requests to be sent.
+		wg.Done()
+		wg.Wait()
+		// Switch out the RangeDescriptorDB.
+		once.Do(func() { rc.TestingSetDB(splitRDB) })
+	}
+
+	transportFn := func(_ context.Context, ba *kvpb.BatchRequest) (*kvpb.BatchResponse, error) {
+		rep := ba.CreateReply()
+		switch ba.ClientRangeInfo.DescriptorGeneration {
+		case 1:
+			waitThenSwitchToSplitDesc()
+			// We have two parallel batches: one with 2 requests and another
+			// with 1 request. Do nothing for the latter.
+			if len(ba.Requests) != 2 {
+				return rep, nil
+			}
+			// For the batch with 2 requests we inject an error to simulate a
+			// stale range cache.
+			rangeMismatchErr := kvpb.NewRangeKeyMismatchError(ctx, nil, nil, &splitDescs[0], nil /* lease */)
+			rangeMismatchErr.AppendRangeInfo(context.Background(), roachpb.RangeInfo{Desc: splitDescs[1], Lease: roachpb.Lease{}})
+			rep.Error = kvpb.NewError(rangeMismatchErr)
+			return rep, nil
+		case 2:
+			// We'll receive a few batches, all with 1 Get. Find the one
+			// targeting keyB and simulate it encountering an error (that could
+			// occur for InitPuts).
+			if len(ba.Requests) != 1 {
+				require.Fail(t, "unexpected number of requests in a batch")
+			}
+			if ba.Requests[0].GetGet().Key.Equal(keyB) {
+				rep.Error = kvpb.NewError(&kvpb.ConditionFailedError{})
+				// This is the zeroth request in a batch with 1 request.
+				rep.Error.SetErrorIndex(0)
+			}
+			return rep, nil
+		default:
+			require.Fail(t, "unexpected desc generation")
+			return nil, nil
+		}
+	}
+
+	cfg := DistSenderConfig{
+		AmbientCtx:        log.AmbientContext{Tracer: tr},
+		Clock:             clock,
+		NodeDescs:         g,
+		RPCContext:        rpcContext,
+		RangeDescriptorDB: initialRDB,
+		TestingKnobs: ClientTestingKnobs{
+			TransportFactory: adaptSimpleTransport(transportFn),
+		},
+		Settings: cluster.MakeTestingClusterSettings(),
+	}
+
+	ds := NewDistSender(cfg)
+	rc = ds.rangeCache
+
+	ba := &kvpb.BatchRequest{}
+	// Issue a BatchRequest such that it touches the original two ranges, and
+	// then the last request gets an error injected.
+	ba.Add(kvpb.NewGet(keyA), kvpb.NewGet(keyC), kvpb.NewGet(keyB))
+	// Inconsistent read because otherwise the batch will ask to be re-sent in a
+	// txn when split.
+	ba.ReadConsistency = kvpb.INCONSISTENT
+
+	_, pErr := ds.Send(ctx, ba)
+	require.NotNil(t, pErr)
+	require.NotNil(t, pErr.Index)
+	// This is the crux of the test - the error should have the position of
+	// Get(keyB) request from the original batch.
+	require.Equal(t, int32(2), pErr.Index.Index)
 }
 
 // Test the following scenario: the DistSender sends a request that results in a

